### PR TITLE
feat(sdk-coin-eth): build from rawTx

### PR DESCRIPTION
**Pull request by mullapudipruthvik**
Ticket: BG-63332

Add support to build from rawTx for transaction of type recoveryWalletDeployment

### CODE REVIEW

File: modules/sdk-coin-polygon/test/unit/transactionBuilder/walletInitialization.ts [hunk 1]
```diff
@@ -141,5 +141,41 @@ describe('Polygon wallet initialization', function () {
       txBuilder.counter(1);
       await txBuilder.build().should.be.rejectedWith('Invalid transaction: missing contract call data field');
     });
+
+    it('should build and sign using txHex', async () => {
```
Comment (sachushaji) [inline modules/sdk-coin-polygon/test/unit/transactionBuilder/walletInitialization.ts:145]: Why are we adding the test case in polygon when we made a change in eth module? can't we write the tests there?

Comment (mullapudipruthvik) [reply modules/sdk-coin-polygon/test/unit/transactionBuilder/walletInitialization.ts]: sure anyway polygon uses eth transaction builder
added in eth tests

File: modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts [hunk 2]
```diff
@@ -404,4 +404,42 @@ describe('Eth Transaction builder wallet initialization', function () {
     const txFromRaw = await txBuiderFromRaw.build();
     should.deepEqual(tx.id, txFromRaw.id);
   });
+
+  describe('Recovery Wallet', function () {
+    it('should build and sign using txHex', async () => {
+      const txBuilder = getBuilder('tpolygon') as TransactionBuilder;
```
Comment (sachushaji) [inline modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts:410]:
```suggestion
      const txBuilder = getBuilder('gteth') as TransactionBuilder;
```

File: modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts [hunk 3]
```diff
@@ -404,4 +404,42 @@ describe('Eth Transaction builder wallet initialization', function () {
...
+      txBuilder.from(txHex);
+      const xpriv =
+        'xprv9s21ZrQH143K2YKSzZa2fv7QTzZ5Ui45ZUzygDwhfeHodwHbWtUUSKqikMKZR9Z751NMekrjXKBykb8mhvoTrKMRodKBvNDH4XKPkccF1K4';
+      txBuilder.sign({ key: xpriv });
+      const signedTx = await txBuilder.build();
+      should.exists(signedTx.toBroadcastFormat());
+    });
+
+    it('succeed when build via contract call', async () => {
+      const txBuilder = getBuilder('tpolygon') as TransactionBuilder;
```
Comment (sachushaji) [inline modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts:421]:
```suggestion
      const txBuilder = getBuilder('gteth') as TransactionBuilder;
```

File: modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts [hunk 4]
```diff
@@ -404,4 +404,42 @@ describe('Eth Transaction builder wallet initialization', function () {
...
+        gasLimit: '10000',
+      });
+
+      const data = testData.SEND_FUNDS_METHOD_CALL;
+      txBuilder.counter(1);
+      txBuilder.data(data);
+      txBuilder.contract('0xd536f4b9f9127a39f19820ca18baac7cd157471f');
+      const tx = await txBuilder.build();
+      const txhex = tx.toBroadcastFormat();
+      const builderFrom = getBuilder('tpolygon') as TransactionBuilder;
```
Comment (sachushaji) [inline modules/sdk-coin-eth/test/unit/transactionBuilder/walletInitialization.ts:438]:
```suggestion
      const builderFrom = getBuilder('gteth') as TransactionBuilder;
```

**Review by sachushaji — APPROVED** [review]

**Pull request merged**